ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -370,16 +373,28 @@ inline double getCacheMaxAgeDuration(std::string cacheControlData) {
 
 void RSkComponentImage::requestNetworkImageData(ImageSource source) {
   auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  networkRequestMutext.lock();
+  if(isRequestinProgress_ == true && remoteCurlRequest_!=nullptr ){
+    sharedCurlNetworking->abortRequest(remoteCurlRequest_);
+    isRequestinProgress_=false;
+  }
+  networkRequestMutext.unlock();
   std::shared_ptr<CurlRequest> remoteCurlRequest = std::make_shared<CurlRequest>(nullptr,source.uri,0,"GET");
-
+  
   folly::dynamic query = folly::dynamic::object();
 
   //Before network request, reset the cache info with default values
   canCacheData_ = true;
   cacheExpiryTime_ = DEFAULT_MAX_CACHE_EXPIRY_TIME;
 
   // headercallback lambda fuction
-  auto headerCallback =  [this, remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+  auto headerCallback =  [this, weakThis = this->weak_from_this(), remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
+    auto isAlive = weakThis.lock();
+    if(!isAlive) {
+       RNS_LOG_WARN("This object is already destroyed. ignoring the completion callback");
+       return 0;
```

Comment:
As this function return type is void returning "0" is not valid or needed. It will be good to correct here & rest of this API.